coderd/database/dbtestutil/db.go

```diff
@@ -51,3 +85,28 @@ func NewDB(t testing.TB) (database.Store, pubsub.Pubsub) {
 
 	return db, ps
 }
+
+// setRandDBTimezone sets the timezone of the database to the given timezone.
+func setDBTimezone(t testing.TB, dbURL, dbname, tz string) {
+	t.Helper()
+
+	sqlDB, err := sql.Open("postgres", dbURL)
+	require.NoError(t, err)
+	defer func() {
+		// The updated timezone only comes into effect on reconnect, so close
+		// the DB after we're done.
+		_ = sqlDB.Close()
+	}()
+
+	// nolint: gosec // This unfortunately does not work with placeholders.
+	_, err = sqlDB.Exec(fmt.Sprintf("ALTER DATABASE %s SET TIMEZONE TO %q", dbname, tz))
```

Comment:
Should we operate on an open/active connection instead?

Reply:
We have to re-open the connection for it to take effect.

Reply:
nit: Oh, I ignored the comment in defer. Maybe it is worth moving it up to the function comment, so it's harder to miss.

TIL: need to reconnect to alter database TZ

Reply:
I think this approach is perfectly fine. We could also update the connection session via `SET TIME ZONE TO '...';`, but setting the db time zone _feels_ more robust, and I'm not 100% sure the effect is the same.